### PR TITLE
Fix issue 3249

### LINKED ISSUE #3249: [BUG] Incorrect time for existing and new transaction
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate problems

### Describe the bug

This is quite a major issue: I added a transaction such as follows:

Screenshot_2024-06-23-15-48-55-747_com ivy wallet

Then when I clicked on the transaction again, the transaction time is now 8 hours behind (which is my time zone GMT+8):

Screenshot_2024-06-23-15-48-59-380_com ivy wallet

This bug occurs on existing transaction where all transactions that I checked now became 8 hours earlier.

Hopefully this is something that will be fixed alongside with the existing one.

### To Reproduce

1. Make sure device time zone is set to non UTC (GMT+0) time zone
2. Create a transaction and note the time
3. Click on the newly created transaction
4. The transaction time is incorrect (a few hours earlier or later depending on the timezone set)

### Expected behavior

New and existing transaction time is correct

### Screenshots

_No response_

### App version

4.6.5 from Google Play (Beta Testing)

### Smartphone

Poco F3, Xiaomi Hyper OS 1.0.2.0, Android 13

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mahmoodharis88**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [ ] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

## What's changed?

Describe with a few bullets **what's new:**
- a
- b
- c

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- a
- b

**In what cases your code won't work?**

- a
- b

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #3249 

> Replace `3249` with the number of the GitHub issue that this PR is fixing. If you've done that correctly, you'll see the issue title linked when previewing your PR description.

> **!Note:** Do **not** link the PR number. Link the number/id of the GitHub Issue from [issues](https://github.com/Ivy-Apps/ivy-wallet/issues).

## Troubleshooting CI failures ❌

GitHub Actions failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt [hunk 1]
```diff
@@ -684,7 +685,7 @@ class EditTransactionViewModel @Inject constructor(
                     dateTime = when {
                         loadedTransaction().dateTime == null &&
                                 dueDate.value == null -> {
-                            timeNowUTC()
+                            timeNowLocal()
```
Comment (ILIYANGERMANOV) [inline screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt:688]: Are you sure that this is sufficient to fix the issue? Can you attach a screen recording where you test with:
- UTC+8
- UTC+0
- UTC-8

Comment (mahmoodharis88) [reply screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt:688]: yes sure.

Comment (mahmoodharis88) [reply screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt:688]: https://github.com/Ivy-Apps/ivy-wallet/assets/42218722/e35ae892-22ca-40d9-9dae-8259561a3ba2


https://github.com/Ivy-Apps/ivy-wallet/assets/42218722/3b1a8fd0-d1b5-44c0-a59a-6a0f10d73783


https://github.com/Ivy-Apps/ivy-wallet/assets/42218722/340aaf0b-5876-4005-90ed-7ebbc8dece91

Comment (ILIYANGERMANOV) [reply screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt:688]: Nice!

**Review by ILIYANGERMANOV — APPROVED** [review]
Awesome. Fix the CI and we can merge 🚀  The Lint errors aren't from this PR but we need to fix them

**Pull request merged**